### DISCUSSION TITLE
Server incorrect default address

### DISCUSSION THREAD

**Discussion by Juanjo-GEx** [directus/directus#15748] [Archive]
### Describe the Bug

The address established by default to start the server is wrong: `Server started at http://0.0.0.0:8055`
The correct address is: `Server started at http://127.0.0.1:8055`

### To Reproduce

The error is reproduced in this video:
[Server bug](https://www.loom.com/share/733a9dd2b1044c42b479521159d21237)

### Errors Shown

```console
No se puede acceder a este sitio web
Es posible que la página web http://0.0.0.0:8055/ esté temporalmente inactiva o que se haya trasladado definitivamente a otra dirección.
ERR_ADDRESS_INVALID
```

**Comment by azrikahar** [comment]
This is because #11465 changed it to display the host it's listening to, which is technically correct as `0.0.0.0`, but yes it isn't correct in terms of reachable URL. This was actually discussed over here: https://github.com/directus/directus/pull/11465#discussion_r802202243

With that said, I think a potential solution would still be the same as what was discussed, which is doing something similar to what [vite](https://github.com/vitejs/vite) and [unjs/listhen](https://github.com/unjs/listhen) (what Nuxt uses under the hood) is doing:

- vite: https://github.com/vitejs/vite/blob/e4c1c6d506069e55eedfb81f8c607ffb3e74bda0/packages/vite/src/node/utils.ts#L824-L875

- listhen: https://github.com/unjs/listhen/blob/a5e84cd9e1581ce53b0cf456e966d055a265a505/src/index.ts#L121-L139

and "translate" `0.0.0.0` to the correct localhost context in terms of URL.

**Reply by Juanjo-GEx** [reply]
People trying to launch their application for the first time will run into the same error and they won't know why, because nothing is specified in the documentation or anywhere else regarding initial configuration.

Having to do a previous configuration is not the best solution, really. I think that some other alternative should be found, so that the default configuration does not present these inconsistency problems.

**Reply by br41nslug** [reply]
Translating `0.0.0.0` to the available addresses seems like a more user friendly approach. The problem i see with that however is  that `0.0.0.0` binds to any and all available addresses and only showing `localhost`/`127.0.0.1` doesnt necessarily work for all deployments 🤔

**Reply by Juanjo-GEx** [reply]
I imagine that you have discussed this technical question in depth. My report is simply made from the point of view of the user, who finds it a bit contradictory that from the console it is shown that the server has been started in one direction when in fact it is not. I was simply commenting on it for user experience rather than as a technical issue.
